src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -0,0 +1,128 @@
+    return {
+        "event_id": event["id"],
+        "span_id": event["trace.span"],
+        "transaction": event["transaction"],
+        "project_id": event["project_id"],
+        "parent_event_id": parent,
+        "is_root": is_root_event,
+    }
+
+
+class OrganizationEventsTraceLightEndpoint(OrganizationEventsEndpointBase):
+    def has_feature(self, organization, request):
+        return features.has(
+            "organizations:trace-view-quick", organization, actor=request.user
+        ) or features.has("organizations:trace-view-summary", organization, actor=request.user)
+
+    def get(self, request, organization, trace_id):
+        if not self.has_feature(organization, request):
+            return Response(status=404)
+
+        try:
+            params = self.get_snuba_params(request, organization)
+        except NoProjects:
+            return Response([])
+
+        with self.handle_query_errors():
+            result = discover.query(
+                selected_columns=[
+                    "id",
+                    "timestamp",
+                    "transaction",
+                    "project_id",
+                    "trace.span",
+                    "trace.parent_span",
+                    'to_other(trace.parent_span, "", 0, 1) AS root',
+                ],
+                # We want to guarantee at least getting the root, and hopefully events near it with timestamp
+                # id is just for consistent results
+                orderby=["-root", "-timestamp", "id"],
+                params=params,
+                query=f"event.type:transaction trace:{trace_id}",
+                limit=100,
+                referrer="api.trace-view.get_ids",
+            )
+            if len(result["data"]) == 0:
+                return Response(status=404)
+
+        event_id = request.GET.get("event_id")
+        if event_id is None:
+            raise ParseError("Only the light trace view is supported at this time")
+
+        if is_root(result["data"][0]):
+            root = result["data"][0]
+        else:
+            root = None
+            logger.warning(
+                "discover.trace-view.root.not-found",
+                extra={"trace": trace_id, "organization": organization},
+            )
+            return Response(status=204)
+
+        # Look for extra roots
+        extra_roots = 0
+        for item in result["data"][1:]:
+            if is_root(item):
+                extra_roots += 1
+            else:
+                break
+        if extra_roots > 0:
+            logger.warning(
+                "discover.trace-view.root.extra-found",
+                extra={"trace": trace_id, "organization": organization, "extra_roots": extra_roots},
+            )
+
+        return Response(self.serialize(result["data"], root, event_id))
+
+    def serialize(self, result, root, event_id=None):
+        parent_map = {item["trace.parent_span"]: item for item in result}
```

Comment:
@wmak @Zylphrex  Would it be an issue if two or more events have the same `trace.parent_span` value?

Reply:
[VIS-665](https://getsentry.atlassian.net/browse/VIS-665) I'll tackle fixing that sometime this sprint